src/main/kotlin/com/terraformation/backend/customer/ProjectService.kt

```diff
@@ -29,4 +45,18 @@ class ProjectService(
       plantingSiteStore.assignProject(projectId, plantingSiteIds)
     }
   }
+
+  fun addInternalUserRole(
+      projectId: ProjectId,
+      userId: UserId,
+      role: ProjectInternalRole? = null,
+      roleName: String? = null
+  ) {
+    val user = userStore.fetchOneById(userId)
+    if (user.globalRoles.isEmpty()) {
+      throw IllegalStateException("Only global users can be added as project internal roles.")
```

Comment:
Kind of a nit and maybe even me being wrong about terminology since the product folks have gone all-in on it, but "global user" isn't really a concept in the code or data model. There are users, and there are global roles (where the term "global" just means the roles aren't scoped to specific organizations) and users can have global roles. If I were writing this, I'd probably make it say something like, "User has no global roles."

But I'm biased because I came up with the term "global role," and "global user" has always kind of rubbed me the wrong way because it sort of implies the underlying user is of a different type. So you can ignore this if you like.

Reply:
Ya I feel conflicted here about the implentation overall, because we might one day add a global role that should _not_ be able to be a project internal user...